client/components/calypso-i18n/index.tsx

```diff
@@ -0,0 +1,204 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { setLocaleData } from '@wordpress/i18n';
+import { useI18n, I18nProvider } from '@automattic/react-i18n';
+import type { Site as SiteStore, User as UserStore } from '@automattic/data-stores';
+import { subscribe, select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getLanguageFile,
+	switchWebpackCSS,
+	getLanguageManifestFile,
+	getTranslationChunkFile,
+} from 'lib/i18n-utils/switch-locale';
+import config from 'config';
+import { getUrlParts } from 'lib/url/url-parts';
+import { getLanguageSlugs } from 'lib/i18n-utils';
+import { USER_STORE } from 'landing/gutenboarding/stores/user'; // TODO: should this be imported from somewhere else?
+
+type User = UserStore.CurrentUser;
+type Site = SiteStore.SiteDetails;
+
+interface AppWindow extends Window {
+	currentUser?: User;
+	i18nLocaleStrings?: string;
+	installedChunks?: string[];
+	__requireChunkCallback__?: {
+		add( callback: Function ): void;
+		getInstalledChunks(): string[];
+	};
+	BUILD_TARGET?: string;
+}
+declare const window: AppWindow;
+
+const DEFAULT_LOCALE_SLUG: string = config( 'i18n_default_locale_slug' );
+const USE_TRANSLATION_CHUNKS: string =
+	config.isEnabled( 'use-translation-chunks' ) ||
+	getUrlParts( document.location.href ).searchParams.has( 'useTranslationChunks' );
+
+export const CalypsoI18nProvider: React.FunctionComponent< {
+	initialLocaleData: any;
+} > = ( { children, initialLocaleData } ) => {
+	const [ cachedLocaleData, setCachedLocaleData ] = React.useState( initialLocaleData );
+
+	useEffect( () => {
+		let isSubscribed = true;
+		getLocale().then(
+			( [ userLocale, { translatedChunks, ...localeData } ]: ( string | any )[] ) => {
+				if ( isSubscribed ) {
+					setCachedLocaleData( localeData );
+				}
+
+				if ( USE_TRANSLATION_CHUNKS ) {
+					setupTranslationChunks( userLocale, translatedChunks );
+				}
+			}
+		);
+		// TODO: handle errors
+		return () => {
+			isSubscribed = false;
+		};
+	}, [] );
+
+	useEffect( () => {
+		setLocaleData( cachedLocaleData );
+	}, [ cachedLocaleData ] );
+
+	useRTLCSS();
+
+	return <I18nProvider localeData={ cachedLocaleData }>{ children }</I18nProvider>;
+};
+
+function useRTLCSS() {
+	const { isRTL } = useI18n();
+	const rtl = isRTL();
+	useEffect( () => {
+		switchWebpackCSS( rtl );
+	}, [ rtl ] );
+	return null;
+}
```

Comment:
Yes, I copied this from there. So... currently there's middleware that detects a RTL language and then changes the webpack CSS, but you're saying that logic may make more sense to be moved into this package. 

It seems like maybe we could leave that for a future improvement rather than the initial work to get a language loader/provider. How does that sound to you?

Reply:
That sounds good to me, we'll want to remove this from here entirely for now and we can let the middleware continue to handle things. Gutenboarding can develop its own handling that can replace the middleware when these components are used across Calypso.